### PR TITLE
#2503  - Make ready for Julia v1.6

### LINKED ISSUE #2503: Make ready for Julia v1.6
See https://github.com/JuliaReach/LazySets.jl/pull/2502

### LINKED ISSUE #2754: Ambiguity in RejectionSampler
v1.6 detects an ambiguity:
```julia
julia> detect_ambiguities(LazySets)
1-element Vector{Tuple{Method, Method}}:
 (LazySets.RejectionSampler(distribution::D, tight::Bool) where D in LazySets at /home/christian/.julia/dev/LazySets/src/Utils/samples.jl:140, LazySets.RejectionSampler(X::LazySet, distribution; tight) in LazySets at /home/christian/.julia/dev/LazySets/src/Utils/samples.jl:152)
```
https://github.com/JuliaReach/LazySets.jl/blob/fc0d95715238e165656033417f25d915a49d7019/src/Utils/samples.jl#L139-L142

https://github.com/JuliaReach/LazySets.jl/blob/fc0d95715238e165656033417f25d915a49d7019/src/Utils/samples.jl#L152

The problem is that `D` can be `LazySet` and `distribution` can be `Bool`. Of course this is not an intended input.

I suggest that we simply define the method and throw an error.

### PR DESCRIPTION
**Pull request by mforets**
Closes #2503.
Closes #2754.

### CODE REVIEW

**Comment by schillic** [comment]
I am a bit worried. Why did the Travis build pass? The majority of files did not get updated yet and in #2507 there were dozens of errors in the doctests.

**Comment by mforets** [comment]
it is cheating on us so we don't switch to github CI :relaxed: 

did it even run the tests? there is no log  for the tests after [this line](https://travis-ci.org/github/JuliaReach/LazySets.jl/jobs/757546686#L925)

**Comment by mforets** [comment]
The bash script https://github.com/JuliaReach/LazySets.jl/issues/2503#issuecomment-772856193 worked like a charm :100:

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        julia-version: ['1.3', '1.5']
+        julia-version: ['1.6']
```
Comment (schillic) [inline .github/workflows/ci.yml]: I would still test in 1.3 (we can exclude the doctests in this case). I just pushed the change. If this works, we can merge.

**Review by schillic — COMMENTED** [review]
:tada:

**Review by schillic — APPROVED** [review]

**Pull request merged**